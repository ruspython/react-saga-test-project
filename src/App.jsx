import { hot } from 'react-hot-loader/root';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Router, Switch, Route } from 'react-router-dom';
import styled, { css, ThemeProvider } from 'styled-components';

import history from 'modules/history';
import theme, { headerHeight } from 'modules/theme';
import { utils } from 'styled-minimal';

import Dashboard from 'routes/Dashboard';
import NotFound from 'routes/NotFound';

import GlobalStyles from 'components/GlobalStyles';

const AppWrapper = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  opacity: 1 !important;
  position: relative;
  transition: opacity 0.5s;
`;

const Main = styled.main`
  min-height: 100vh;
  padding: ${utils.px(headerHeight)} 0 0;
`;

export class App extends React.Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
  };
  render() {
    return (
      <Router history={history}>
        <ThemeProvider theme={theme}>
          <AppWrapper>
            <Main>
              <Switch>
                <Route
                  path="/"
                  exact
                  render={props => (<Dashboard {...props} />)}
                />
                <Route component={NotFound} />
              </Switch>
            </Main>
            <GlobalStyles />
          </AppWrapper>
        </ThemeProvider>
      </Router>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps() {
  return {};
}

export default hot(connect(mapStateToProps)(App));
