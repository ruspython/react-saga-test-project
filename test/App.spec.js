import React from 'react';

import { App } from 'App';

const mockDispatch = jest.fn();

const props = {
  app: {
    alerts: [],
  },
  dispatch: mockDispatch,
  user: {
    isAuthenticated: false,
  },
};

function setup(ownProps = props) {
  return shallow(<App {...ownProps} />, { attachTo: document.getElementById('react') });
}

describe('App', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

});
