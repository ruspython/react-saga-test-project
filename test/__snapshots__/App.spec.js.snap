// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly 1`] = `
<Router history="{{...}}">
  <ThemeProvider theme="{{...}}">
    <AppWrapper>
      <Main>
        <Switch>
          <Route path="/"
                 exact="{true}"
                 render="{[Function:"
                 render]}
          >
          </Route>
          <Route component="{[Function:"
                 notfound]}
          >
          </Route>
        </Switch>
      </Main>
      <_default>
      </_default>
    </AppWrapper>
  </ThemeProvider>
</Router>
`;
